test/test_rpc.ts

```diff
@@ -71,4 +75,50 @@ describe("Rpc", () => {
     });
 
   });
+
+  describe("checker", () => {
+
+    it("should allow calling methods that exist", async () => {
+      const rpc = new Rpc(defaults);
+      rpc.start((msg) => rpc.receiveMessage(msg));
+      rpc.registerImpl<ICalc>("calc", new Calc(), checkersForICalc);
+      const stub = rpc.getStub<ICalc>("calc", checkersForICalc);
+      assert.equal(await stub.add(4, 5), 9);
+    });
+
+    it("should forbid calling methods that do not exist", async () => {
+      const rpc = new Rpc(defaults);
+      rpc.start((msg) => rpc.receiveMessage(msg));
+      rpc.registerImpl<ICalc>("calc", new Calc(), checkersForICalc);
+      const stub = rpc.getStub<ICalc>("calc", checkersForICalc);
+      // "any" cast needed to avoid typescript catching the error
+      assert.throws(() => (stub as any).additionify(4, 5), /is not a function/);
+    });
+
+    it("should catch missing methods at typed stub", async () => {
+      const rpc = new Rpc(defaults);
+      const stub = rpc.getStub<ICalc>("calc", checkersForICalc);
+      assert(stub);
+      // "any" cast needed to avoid typescript catching the error
+      assert.throws(() => (stub as any).additionify(4, 5), /is not a function/);
+    });
+
+    it("should catch missing methods at implementation for untyped stub", async () => {
+      const rpc = new Rpc(defaults);
+      rpc.start((msg) => rpc.receiveMessage(msg));
+      rpc.registerImpl<ICalc>("calc", new Calc(), checkersForICalc);
+      const stub = rpc.getStub<ICalc>("calc");
+      // "any" cast needed to avoid typescript catching the error
+      await assert.isRejected((stub as any).additionify(4, 5), /Unknown method/);
+    });
+
+    it("should catch bad types at implementation", async () => {
+      const rpc = new Rpc(defaults);
+      rpc.start((msg) => rpc.receiveMessage(msg));
+      rpc.registerImpl<ICalc>("calc", new Calc(), checkersForICalc);
+      const stub = rpc.getStub<ICalc>("calc");
+      // "any" cast needed to avoid typescript catching the error
+      await assert.isRejected((stub as any).add("hello", 5), /not a number/);
```

Comment:
Maybe add also a couple more lines here like `add(1, 2, 3)` and `add()` ?

Reply:
Hmm `add(1, 2, 3)` ends up evaluating to `3`.  `add(100, 40, 3)` evaluates to `140`.  So we're not noticing that there's a surplus argument.

Reply:
@dsagal is ignoring surplus arguments expected? It is very javascripty but not so typescripty.  I see that ts-interface-checker has the notion of strict and non-strict checks.

Reply:
Oh I see a discussion at https://github.com/gristlabs/ts-interface-checker#strict-checking